src/com/chrome/apis/Controller.js

```diff
@@ -0,0 +1,299 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+foam.CLASS({
+  package: 'foam.u2.md',
+  name: 'DAOController',
+  extends: 'foam.u2.DAOController',
+
+  requires: [
+    'foam.u2.md.Button',
+    'foam.u2.md.DAOCreateController',
+    'foam.u2.md.DAOUpdateController',
+    'foam.u2.md.Toolbar',
+    'foam.u2.md.ToolbarContainer'
+  ],
+
+  methods: [
+    function initE() {
+      // TODO(braden): Once toPropertyE is generalized to actions etc., use that
+      // here if practical.
+      var list = this.listFactory$f({
+        rowFactory: this.rowFactory,
+        data: this.data
+      }, this);
+
+      list.rowClick.sub(this.rowClick);
+      this.add(list);
+
+      var fab = this.Button.create({
+        action: this.NEW_ITEM,
+        data: this,
+        icon: 'add',
+        type: 'fab'
+      }).cssClass(this.myCls('fab'));
+      this.add(fab);
+    },
+
+    function buildCreateController(X) {
+      var toolbar = this.Toolbar.create();
+      var Y = X.createSubContext({ toolbar: toolbar });
+      return this.ToolbarContainer.create({
```

Comment:
Except for the contexts, yes, that would work fine. The vital part is that the `toolbar` be in the context before the `this.SUPER(Y)` call, so that the buttons get added to the correct toolbar.